test/lib/SchemaConnector-test.js

```diff
@@ -0,0 +1,301 @@
+
+      let expectedAccessToken = 'UNDEFINED';
+      let expectedCallbackAuthentication = {};
+      let expectedCallbackUrls = {};
+
+      schemaConnector.callbackAccessHandler(async (accessToken, callbackAuthentication, callbackUrls) => {
+        expectedAccessToken = accessToken;
+        expectedCallbackAuthentication = callbackAuthentication;
+        expectedCallbackUrls = callbackUrls;
+      });
+
+      let tokenUrl = 'UNDEFINED';
+      let tokenCode = 'UNDEFINED';
+
+      const stub = sinon.stub(AccessTokenRequest.prototype, 'getCallbackToken').callsFake(async function(url, code) {
+        tokenUrl = url;
+        tokenCode = code;
+        return new Promise((resolve, reject) => {
+          resolve({
+            "headers": {
+              "schema": "st-schema",
+              "version": "1.0",
+              "interactionType": "accessTokenResponse",
+              "requestId": "E68C91AB-0ED7-48B2-8FFC-036282FBD607"
+            },
+            "callbackAuthentication": {
+              "tokenType": "Bearer",
+              "accessToken": "eyJhbGciOiJIUzM4NCJ9.MDcyNmRlNDYtNTIzYS00MjlmLWI4MTktODE5NDViODY1MTU1.aLkqFJFmzbrWQNcqsemE_ZHRW8n1Oi1Xrvo_4aB_q7vsUKk03z4rpi-XbyLQ285o",
+              "refreshToken": "eyJhbGciOiJIUzM4NCJ9.ZWQ0YzM1ODQtNjA3My00MWViLTkxZTctNDU3ZWNhZmQ2NThm.IAX_H7mFZMHF2PVguvPxHMU8I2-UqDLmszVTDTKWnLVWD64OKm3pYGXWCcFqbQvZ",
+              "expiresIn": 86400
+            }
+          })
+        })
+      });
+
+      const response = await schemaConnector.handleMockCallback({
+        "headers": {
+          "schema": "st-schema",
+          "version": "1.0",
+          "interactionType": "grantCallbackAccess",
+          "requestId": "8C58E0CD-386F-4BBB-80B9-B28A7FF8040F"
+        },
+        "authentication": {
+          "tokenType": "Bearer",
+          "token": "ACCT-HCtR4Q"
+        },
+        "callbackAuthentication": {
+          "grantType": "authorization_code",
+          "scope": "callback-access",
+          "code": "eyJhbGciOiJIUzM4NCJ9.MTFlZTdlMTMtZjQ1My00YTI0LWI2ZTktZTY4MGU4ZDU4NDIx.pVSGwU1hWKFuA4Snzk_vacZH1luHDQgGGaClgPZ4vpULPu35aXtBp6aURNYElxyl",
+          "clientId": "xxxx"
+        },
+        "callbackUrls": {
+          "oauthToken": "https://c2c-us.smartthings.com/oauth/token",
+          "stateCallback": "https://c2c-us.smartthings.com/device/events"
+        }
+      });
+
+      tokenUrl.should.equal('https://c2c-us.smartthings.com/oauth/token');
+      tokenCode.should.equal('eyJhbGciOiJIUzM4NCJ9.MTFlZTdlMTMtZjQ1My00YTI0LWI2ZTktZTY4MGU4ZDU4NDIx.pVSGwU1hWKFuA4Snzk_vacZH1luHDQgGGaClgPZ4vpULPu35aXtBp6aURNYElxyl');
+
+      expectedAccessToken.should.equal('ACCT-HCtR4Q');
+      expectedCallbackAuthentication.tokenType.should.equal('Bearer');
+      expectedCallbackAuthentication.accessToken.should.equal('eyJhbGciOiJIUzM4NCJ9.MDcyNmRlNDYtNTIzYS00MjlmLWI4MTktODE5NDViODY1MTU1.aLkqFJFmzbrWQNcqsemE_ZHRW8n1Oi1Xrvo_4aB_q7vsUKk03z4rpi-XbyLQ285o');
```

Comment:
Maybe we should assign this long token into a constant and use that here?

Reply:
:thumbsup: